luomu-libpcap/src/functions.rs

```diff
@@ -372,6 +366,83 @@ pub fn pcap_open_dead() -> Result<PcapT> {
     })
 }
 
+/// open a file to which to write packets
+///
+/// `pcap_dump_fopen()` is called to write data to an existing open file; The
+/// stream is assumed to be at the beginning of a file that has been newly
+/// created or truncated, so that writes will start at the beginning of the
+/// file.
+///
+/// <https://www.tcpdump.org/manpages/pcap_dump_open.3pcap.html>
+//
+// Rust compiler doesn't require it, but we want to take `&mut PcapDumper`
+// because we write into a file pointed by `PcapDumper`.
+#[allow(clippy::needless_pass_by_value)]
+pub fn pcap_dump_fopen(pcap_t: &PcapT, file: &mut std::fs::File) -> Result<PcapDumper> {
+    trace!("pcap_dump_fopen({:p}, {:?})", pcap_t.pcap_t, file);
+    let mode = b"wb\0";
+
+    let filedesc = unsafe { libc::fdopen(file.as_raw_fd(), mode.as_ptr() as *const libc::c_char) };
+    if filedesc.is_null() {
+        return Err(Error::IO(std::io::Error::last_os_error()));
+    }
+
+    let ret = unsafe { libpcap::pcap_dump_fopen(pcap_t.pcap_t, filedesc as *mut [u64; 19]) };
```

Comment:
where does the `[u64; 19]` come from. Clarifying comment would be nice.

Reply:
This cast is no more.